package/MDAnalysis/analysis/density.py

```diff
@@ -434,6 +444,11 @@ def _single_frame(self):
         self._grid += h
 
     def _conclude(self):
+        if len(self._atomgroup)==0:
+            msg = ("No atoms in selection over whole trajectory")
+            warnings.warn(msg)
+            logger.warning(msg)
+            return
```

Comment:
I wonder if three duplications of the same code block is enough to justify abstraction to a single i.e., function. Maybe borderline if the `return` has to remain outside of the abstraction I suppose.

On a more basic level, just asking--the code will still produce the original failure/exception (is that what it was doing?) and this will simply add a warning to clarify?

Reply:
This might sound very primitive but my logic was to return out of the function and warn the user of no atoms until the atomgroup gets updated with atleast one atom. In the Issue, the user wanted to avoid the program crashing and would provide a dummy atom to keep the density frames generating. I tried to fix this by preventing the crash at the initializing, frame generation and conclusion. Perhaps a more efficient way is to make a function to check for no atoms and return.

Reply:
Again, my understanding of the function process may not be very accurate so I apologize for any incorrect arguments I make :P